src/modules/subject/subjects.routes.js

```diff
@@ -0,0 +1,17 @@
+const router = require('express').Router()
+const subjectController = require('./subject.controller')
+const asyncWrapper = require('~/middlewares/asyncWrapper')
+const idValidation = require('~/middlewares/idValidation')
+const { authMiddleware } = require('~/middlewares/auth')
+
+router.use(authMiddleware)
+router.param('id', idValidation)
+
+router.get('/', asyncWrapper(subjectController.subjectsFind))
+router.get('/:id', asyncWrapper(subjectController.subjectFindById))
+router.get('/categories/:id', asyncWrapper(subjectController.subjectsFindByCategoryId))
+router.post('/', asyncWrapper(subjectController.subjectCreate))
+router.put('/:id', asyncWrapper(subjectController.subjectUpdate))
+router.delete('/:id', asyncWrapper(subjectController.subjectDelete))
```

Comment:
Are there any restrictions for the user role? According requirements only admins should be able create, upd or delete subjects...

Reply:
yes, need to add restrict To(ADMIN)